### DISCUSSION TITLE
How to debug font issues?

### DISCUSSION THREAD

**Discussion by AckslD** [nvim-lualine/lualine.nvim#657] [Q&A] [answered]
I recently updated my system and the font characters used by lualine seems to be broken and it looks as follows now:

![gscreenshot_2022-04-19-150751](https://user-images.githubusercontent.com/23341710/164011276-5b97ae9c-755e-4468-a840-353ab5a82a18.png)

Any advice on how to debug what's going wrong? I haven't installed/uninstalled any fonts on my system, nor change my terminal settings. It seems to have been caused by an update to either kitty, neovim or lualine.

**Comment by shadmansaleh** [comment, marked as answer]
Are the sections suppose to be colored or not ? I mean are you using a gray scale type theme or is it kitty not drawing correct colors . It'd be easier to figure out if you provide your config & theme. 

Can you cd into your lualine installation open nvimand run the following commad ?
```vim
:lua =require'tests.statusline':snapshot()
```
What does the output say ?

**Reply by AckslD** [reply]
Thanks for you response @shadmansaleh! The above command gives:
```lua
statusline:expect([===[                                                                                                                                                              
highlights = {
    1: lualine_a_normal = { bg = "#a89984", bold = true, fg = "#282828" }
    2: lualine_transitional_lualine_a_normal_to_lualine_b_normal = { bg = "#504945", fg = "#a89984" }
    3: lualine_b_normal = { bg = "#504945", fg = "#ebdbb2" }
    4: lualine_transitional_lualine_b_normal_to_lualine_c_normal = { bg = "#3c3836", fg = "#504945" }
    5: lualine_c_normal = { bg = "#3c3836", fg = "#a89984" }
    6: lualine_x_summary_clean_normal = { bg = "#3c3836", fg = "#b8bb26" }
    7: lualine_x_summary_clean_0_normal = { bg = "#3c3836", fg = "#fb4934" }
    8: lualine_transitional_lualine_b_normal_to_lualine_x_summary_clean_0_normal = { bg = "#3c3836", fg = "#504945" }
}
|{1: NORMAL }
{2:}
{3:  master }
{4:}
{5: [No Name] }
{5:                                                        }
{5: ]q }
{5:  }
{6: ✔53 }
{7:✘3 }
{8:}
{3: 100% }
{2:}
{1:   0:1  }|
]===])
```

The sections were indeed colored before but now seem to be transparent. Not sure what changed.

**Reply by shadmansaleh** [reply]
> The sections were indeed colored before but now seem to be transparent. Not sure what changed.

That's the main issue here. From the output you've provided lualine is setting up highlight groups correctly . After that it's neovims job to tell terminal what to draw in what color and terminal is suppose to do the actual drawing . 

Are you only missing colors in only statusline or in rest of neovim & other terminal programs ? Can you try on a different terminal or revert kitty ?

**Reply by AckslD** [reply]
Thanks @shadmansaleh! After trying do downgrade kitty, neovim and some other plugins I tracked it down to a recent change to `gruvbox.nvim` which I'm using. See:
https://github.com/ellisonleao/gruvbox.nvim/issues/112

**Reply by shadmansaleh** [reply]
That's actually strange as they don't seem to provide a lualine theme. Are you loading lualine before or after your colorscheme ? 

Also does `:hi lualine_a_normal` say cleared ?

**Reply by AckslD** [reply]
> Also does :hi lualine_a_normal say cleared ?

Yes it says:
```
lualine_a_normal xxx cleared
```

> Are you loading lualine before or after your colorscheme ?

Hmm, not sure actually, both are loaded on `VimEnter`. Should one be loaded before the other?

**Reply by shadmansaleh** [reply]
> Should one be loaded before the other?

lualine must be loaded after colorscheme . By after means you need to set your colorscheme before loading lualine.

**Reply by fitrh** [reply]
@AckslD if you are using packer, for lazy loading, as long as you load your color scheme using `colorscheme` command, I suggest to load lualine using `ColorSchme` event to ensure that it loaded after the color scheme is set

**Reply by AckslD** [reply]
Thanks a lot both of you! Adding `after = 'gruvbox.nvim'` to my packer settings for `lualine.nvim` worked :+1: 

For some reason, lazy loading on `ColorScheme` didn't, but at least it's working now :)

**Comment by fitrh** [comment]
@AckslD have you tried to disabling the inverse setting (`vim.g.gruvbox_inverse = false`?)

**Reply by AckslD** [reply]
Thanks for the suggestion @fitrh, it seems to make no difference though.

**Reply by fitrh** [reply]
IDK, It still incomplete, but it seem that author will [migrate](https://github.com/ellisonleao/gruvbox.nvim/issues/96) configuration into lua `setup` style and diverge from original gruvbox, as you can [see](https://github.com/ellisonleao/gruvbox.nvim/blob/028302225e19eb203ebb6876b689cc0ce2631169/lua/gruvbox/base.lua#L24), in the default configuration table, it says `italicize_comments` but [highlight declaration](https://github.com/ellisonleao/gruvbox.nvim/blob/028302225e19eb203ebb6876b689cc0ce2631169/lua/gruvbox/base.lua#L284) is still using `gruvbox_italic_comments`
